### DISCUSSION TITLE
SX1272 CAD problems

### DISCUSSION THREAD

**Discussion by squarecode** [Lora-net/LoRaMac-node#1027] [Q&A]
Hey, I'm trying to get the channel activity detection on the sx1272 working. My boards can Rx and Tx without any problems. I have one board set up to periodically tx a test packet. It consists of the preamble (1020 + 4 Symbols added by the hardware as the datasheet states) and 64 bytes of test data. I can rx the packet and the correct contents of the packet with another board. I now have set another board to start cad periodically:
```
...
// init of the radio as for the rx/tx nodes
...

Radio.Standby();

while(1)
{
   Radio.StartCad();
   while(!((SX1272Read( REG_LR_IRQFLAGS ) & RFLR_IRQFLAGS_CADDONE ) == RFLR_IRQFLAGS_CADDONE))
   {
   }
   // at this point the CadDone bit is set
   debug(SX1272Read( REG_LR_IRQFLAGS )); // output of the current irq register state to verify
   if((SX1272Read( REG_LR_IRQFLAGS ) & RFLR_IRQFLAGS_CADDETECTED ) == RFLR_IRQFLAGS_CADDETECTED)
    {
        debug( "DETECTED");
    }

   SX1272Write( REG_LR_IRQFLAGS, RFLR_IRQFLAGS_CADDONE | RFLR_IRQFLAGS_CADDETECTED); // reset irq

}
```

The interrupt fires as expected and the CadDone bit is set after a short period. However the CadDetected has never catched once for all spreading factors and preamble length that I've tried. What am I missing here?

I have tried hardware irq with the SX1272OnDio3Irq() and DIO Mapping 0 and they work fine too. But for debugging purposes I switched to a software polling approach for now.

Thanks for your help!

**Comment by mluis1** [comment]
On the following forum there are some topics discussing the same kind of questions.

https://forum.lora-developers.semtech.com/

I would recommend to go there in order to get better support on Semtech radios usage.
